File: dsa/lab11/p2.py
def mergesort(a,low,high):
	if(low<high):
		mid=int((low+high)/2)
		mergesort(a,low,mid)
		mergesort(a,mid+1,high)
		merge(a,low,mid,high)
def merge(a,low,mid,high):
	a1=[]
	a2=[]
	# a1[i]=a[low:mid+1]
	# a2[i]=a[mid+1:high]
	for i in range(low,mid+1):
		a1.append(a[i])
	for i in range(mid+1,high+1):
		a2.append(a[i])
	print(a1,a2)
	i=0
	j=0
	k=low
	while (i<len(a1) and j<len(a2)):
		if a1[i]<a2[j]:
			a[k]=a1[i]
			i+=1
			k+=1
		else:
			a[k]=a2[j]
			j+=1
			k+=1
	while(i<len(a1)):
		a[k]=a1[i]
		k+=1
		i+=1
	while(j<len(a2)):
		a[k]=a2[j]
		k+=1
		j+=1	
def main():
	a=[56,-7,5,34,0,7,23,9]
	b=[0,1,6,2]
	mergesort(a,0,7)
	mergesort(b,0,3)
	print(b)
	print(a)
if __name__ == '__main__':
	main()

File: Sem4/daa/lab2/p2.c
#include<stdio.h>
#include<stdlib.h>
// int peak(int a[],int low,int high);
int peak(int *a,int low,int high)
{   if(low==high)return a[low];
	if(low<=high)
	{
		int mid=(low+high)/2;
		if(mid-1<0)return a[high];
		if(a[mid]>a[mid-1])return peak(a,mid,high);
		else
		{   
			if(a[mid-1]>a[mid])return peak(a,low,mid-1);
			else return a[mid];
		}
	}
	return -1;
}
void  main()
{
	int a[]={2,4,6,8,7,5,3};
	int b[]={10, 12, 8, 4 , -3, -15};
	int a1=peak(a,0,6);
	printf("%d\n",a1);
	printf("%d",peak(b,0,5));
}